src/features/auth/Register.tsx

```diff
@@ -0,0 +1,57 @@
+import Box from "@mui/material/Box";
+import Typography from "@mui/material/Typography";
+import FormControl from "@mui/material/FormControl";
+import RegisterForm from "./register-form/RegisterForm";
+import { FunctionComponent, useState, useCallback } from "react";
+import registerService from "features/auth/services/register.service";
+
+import useRedirectWhenRegistered from "./use-redirect-when-registered";
+
+const Register: FunctionComponent = () => {
+  const { loading, onSubmit } = useRegister();
+  return (
+    <Box>
+      <Typography>Register</Typography>
+      <FormControl margin="normal">
+        <RegisterForm disabled={loading} onSubmit={onSubmit} />
+      </FormControl>
+    </Box>
+  );
+};
+
+function useRegister() {
+  useRedirectWhenRegistered();
+
+  const [loading, setLoading] = useState(false);
+
+  const onSubmit = useCallback(
+    async (
+      name: string,
+      email: string,
+      password: string,
+      confirmPassword: string
+    ) => {
```

Comment:
El tio bob lo dice, usa un objeto cuando son mas de 2 argumentos en una funcion
```suggestion
    async ({
      name,
      email,
      password,
      confirmPassword
    }: RegisterPayload) => {
```